activerecord/lib/active_record/associations/builder/has_one.rb

```diff
@@ -26,5 +32,33 @@ def self.define_validations(model, reflection)
         model.validates_presence_of reflection.name, message: :required
       end
     end
+
+    def self.touch_record(o, name, touch, touch_method) # :nodoc:
+      record = o.send name
+
+      return unless record && record.persisted?
+
+      if touch != true
+        record.send(touch_method, touch)
+      else
+        record.send(touch_method)
+      end
+    end
+
+    def self.add_touch_callbacks(model, reflection)
+      name  = reflection.name
+      touch = reflection.options[:touch]
+
+      callback = lambda { |changes_method| lambda { |record|
```

Comment:
`changes_method` is not used.

Reply:
@kamipo Should we remove `lambda { |changes_method| }` here. If we do that I am not able to get `record` to invoke `HasOne.touch_record`.

Reply:
https://github.com/rails/rails/blob/6b5ad4979ed2ba268755c74c804a2e22eee1d5b8/activerecord/lib/active_record/associations/builder/belongs_to.rb#L110-L116

Reply:
Thanks, @kamipo . That worked. Fixed it.